modules/tfm/tfm/boards/common/tfm_hal_platform.c

```diff
@@ -27,20 +31,22 @@ enum tfm_hal_status_t tfm_hal_platform_init(void)
 #if defined(TFM_PARTITION_CRYPTO)
```

Comment:
Let's move crypto initialization into it's own function so that tfm_hal_platform_init reads like:

```
enum tfm_hal_status_t status;
	status = tfm_hal_platform_common_init();
	if (status != TFM_HAL_SUCCESS) {
		return status;
	}

#if defined(TFM_PARTITION_CRYPTO)
	status = crypto_init();
	if (status != TFM_HAL_SUCCESS) {
		return status;
	}
#endif

return TFM_HAL_SUCCESS;
```

When practical, a function should do things at a consistent abstraction level. (either only low-level details, or only high level details). Here we do high-level, and then low-level.